xds/server.go

```diff
@@ -315,11 +240,18 @@ func (s *GRPCServer) GracefulStop() {
 func routeAndProcess(ctx context.Context) error {
 	conn := transport.GetConnection(ctx)
 	cw, ok := conn.(interface {
-		VirtualHosts() []xdsresource.VirtualHostWithInterceptors
+		RoutingConfiguration() xdsresource.RoutingConfiguration
 	})
 	if !ok {
 		return errors.New("missing virtual hosts in incoming context")
 	}
+
+	rc := cw.RoutingConfiguration()
+	if rc.Err != nil { // Error out at routing l7 level, represents an nack before usable route configuration or resource not found for RDS or error combining LDS + RDS (Shouldn't happen).
+		logger.Errorf("RPC on connection with xDS Configuration error: %v", rc.Err)
```

Comment:
Might want to wrap this in a verbosity level 2, and change this to INFO logs. This could well cause log spam on a real production server without the verbosity check.

Reply:
Done.